src/components/split-button/split-button-rtl.test.tsx

```diff
@@ -0,0 +1,912 @@
+
+    const toggle = screen.getByRole("button", { name: "Show more" });
+    await user.click(toggle);
+    const childButton = screen.getByRole("button", { name: "Single Button" });
+
+    expect(childButton).toBeInTheDocument();
+
+    await waitFor(async () => {
+      await user.click(document.body);
+
+      expect(screen.queryByRole("list")).not.toBeInTheDocument();
+    });
+  });
+
+  it("should hide the additional buttons when a mouseout event detected on the toggle button", async () => {
+    const user = userEvent.setup();
+    render(<SplitButton text="Main">{singleButton()}</SplitButton>);
+
+    const toggle = screen.getByRole("button", { name: "Show more" });
+    await user.hover(toggle);
+    const childButton = screen.getByRole("button", { name: "Single Button" });
+
+    expect(childButton).toBeInTheDocument();
+
+    await waitFor(async () => {
+      await user.unhover(toggle);
+
+      expect(screen.queryByRole("list")).not.toBeInTheDocument();
+    });
+  });
+
+  it("should not hide the additional buttons when a mouseout event detected but the toggle button is focused", async () => {
+    const user = userEvent.setup();
+    render(<SplitButton text="Main">{singleButton()}</SplitButton>);
+
+    const toggle = screen.getByRole("button", { name: "Show more" });
+    await user.click(toggle);
+    const childButton = screen.getByRole("button", { name: "Single Button" });
+
+    expect(childButton).toBeInTheDocument();
+    await user.unhover(toggle);
+    expect(childButton).toBeInTheDocument();
+  });
+
+  it("should hide the additional buttons when a 'Escape' keydown event detected and focus is within component", async () => {
+    const user = userEvent.setup();
+    render(<SplitButton text="Main">{singleButton()}</SplitButton>);
+
+    const { button1 } = await openMultipleAdditionalButtons(user, true);
+
+    expect(button1).toBeInTheDocument();
+
+    await waitFor(async () => {
+      await user.keyboard("{Escape}");
+
+      expect(screen.queryByRole("list")).not.toBeInTheDocument();
+    });
+  });
+
+  it("should hide the additional buttons when a 'Escape' keydown event detected and focus is not within component", async () => {
+    const user = userEvent.setup();
+    render(<SplitButton text="Main">{singleButton()}</SplitButton>);
+
+    const toggle = await screen.findByRole("button", { name: "Show more" });
+    await user.hover(toggle);
+    const button1 = await screen.findByRole("button", {
+      name: "Single Button",
+    });
+
+    expect(button1).toBeInTheDocument();
+    await waitFor(async () => {
+      await user.keyboard("{Escape}");
+
+      expect(screen.queryByRole("list")).not.toBeInTheDocument();
+    });
+  });
+
+  it("should render with the correct styles when 'align' prop is 'right'", async () => {
+    render(
+      <SplitButton text="Main" align="right">
+        {singleButton()}
+      </SplitButton>
+    );
+
+    openMultipleAdditionalButtons(userEvent.setup(), true);
```

Comment:
I don't think a helper function that clicks an returns the children is so crazy that it's unreadable, the name is definitely incorrect (I'd probably split them on further thought) but I'm personally okay with small functions that are named intuitively: we want to avoid really complex test setups such as iterating over multi-dimensional arrays etc or anything that has a complex or non-standardised render function

Reply:
I've removed it after further reflection 😆